src/client.ts

```diff
@@ -1,20 +1,41 @@
 import { RequestInit } from 'node-fetch'
 import { Handler } from './handler/handler'
 import { deserializeResult, serializeFunc } from './serialize'
-import { IRpc } from './type'
+import { IError, IRpc } from './type'
 import { DEFAULT_PATH } from './util'
 import { timedFetch, Fetch } from './timedFetch'
 
+/**
+ * @param errorCallback If present it will be called when client-side errors
+ * happens, since these errors are not handled by default. If this callback
+ * is not provided then errors are going to be thrown instead.
+ */
 export function makeClient<Impl extends IRpc<Impl>>(
   connector: Connector,
+  /**
+   * Allow users to handle errors more gracefully by letting them return
+   * an IError instead of throwing them.
+   *
+   * @param error The error, can be of any type (usually is a JSON detailing)
+   * the incident.
+   */
+  errorCallback?: (error?: unknown) => Promise<IError> | IError
```

Comment:
Let's call this `handleError: boolean | (x: Error, name: string, arg: <something>)`.